### PR TITLE
[Aot] Deprecate element shape and field dim for AOT symbolic args

### PR DESCRIPTION
**Pull request by turbo0628**
Issue: #6572 

* Fix using dtype for symbolic args in AOT usages
* Cleanup all the usages of `field_dim` and `element_shape` in examples and tests

### CODE REVIEW

File: python/taichi/graph/_graph.py [hunk 1]
```diff
@@ -97,25 +99,39 @@ def run(self, args):
...
         channel_format=None,
         shape=(),
         num_channels=None):
+    if field_dim is not None:
+        if ndim != 0:
+            raise TaichiRuntimeError(
+                'field_dim is deprecated, please do not specify field_dim and ndim at the same time.'
+            )
+        warnings.warn(
+            "The field_dim argument for ndarray will be deprecated in v1.5.0, use ndim instead.",
```
Comment (lin-hitonami) [inline python/taichi/graph/_graph.py:114]: Should we add an test to `test_deprecation.py`?

Comment (turbo0628) [reply python/taichi/graph/_graph.py:114]: Thanks, added the deprecation tests!

File: tests/cpp/aot/python_scripts/comet_aot.py [hunk 2]
```diff
@@ -40,11 +40,7 @@
 count = ti.field(ti.i32, ())
 img = ti.field(ti.f32, (res, res))
 
-sym_arr = ti.graph.Arg(ti.graph.ArgKind.NDARRAY,
-                       'arr',
-                       ti.f32,
-                       field_dim=3,
-                       element_shape=())
+sym_arr = ti.graph.Arg(ti.graph.ArgKind.NDARRAY, 'arr', ti.f32, ndim=3)
```
Comment (ailzhang) [inline tests/cpp/aot/python_scripts/comet_aot.py:43]: nit: shall  we spread `dtype=ti.f32` across the tests to encourage users to specify kw for better clarity? ;)

**Review by ailzhang — COMMENTED** [review]
Thanks!

File: tests/cpp/aot/python_scripts/comet_aot.py (hunk 2 above)
Comment (turbo0628) [reply tests/cpp/aot/python_scripts/comet_aot.py:43]: Have updated all occurrences with kw dtype inside tests and examples!

File: python/taichi/graph/_graph.py [hunk 3]
```diff
@@ -130,7 +146,11 @@ def Arg(tag,
                             channel_format=channel_format,
                             num_channels=num_channels,
                             shape=shape)
-    return _ti_core.Arg(tag, name, dtype, field_dim, element_shape)
+    if len(element_shape) > 0:
```
[the remark this answers is not in the document]
Comment (turbo0628) [reply python/taichi/graph/_graph.py:149]: Thanks, it's done :)

**Review by ailzhang — APPROVED** [review]
Thanks！

**Pull request merged**